### PR TITLE
[IMP] data cleaning: recycle records

### PR DESCRIPTION
**Pull request by samueljlieber**
Task: [#4009128](https://www.odoo.com/odoo/project/3835/tasks/4009128?cids=3)

This PR adds Recycle records content to the Data Cleaning doc.

Should be merged after https://github.com/odoo/documentation/pull/9866 up to master.

### CODE REVIEW

**Comment by samueljlieber** [comment]
Hi @jero-odoo! Would you mind doing a first-round peer review on this PR, TIA!

File: content/applications/productivity/data_cleaning.rst [hunk 1]
```diff
@@ -211,9 +211,117 @@ Click on the :icon:`fa-clone` :guilabel:`Duplicates` smart button to manage thes
...
+
+By default, the :guilabel:`All` option is selected. Records are displayed in the list view, with the
+following columns:
+
+- :guilabel:`Record ID`: the ID of the original record.
+- :guilabel:`Record Name`: the name or title of the original record.
+
+Select a specific rule in the :guilabel:`RULE` sidebar to filter the duplicate records.
+
+To recycle records, click on the :icon:`fa-check` :guilabel:`Validate` button on the row of the
```
Comment (jero-odoo) [inline content/applications/productivity/data_cleaning.rst:235]:
```suggestion
To recycle records, click the :icon:`fa-check` :guilabel:`Validate` button on the row of the
```

File: content/applications/productivity/data_cleaning.rst [hunk 2]
```diff
@@ -211,9 +211,117 @@ Click on the :icon:`fa-clone` :guilabel:`Duplicates` smart button to manage thes
...
 On the :guilabel:`Field Recycle Records` dashboard (:menuselection:`Data Cleaning app --> Recycle
 Records`), Odoo detects records that can be archived or deleted, by matching conditions within the
-records set by the recycle record's rules.
+records set by the :ref:`recycle record's rules <data_cleaning/recylce-rule>`.
+
+.. image:: data_cleaning/data-cleaning-recycle.png
+   :align: center
+   :alt: Field Recycle Records dashboard in the Data Cleaning application.
+
+The :guilabel:`RULE` sidebar lists each of the active recycle record rules, and displays the total
```
Comment (jero-odoo) [inline content/applications/productivity/data_cleaning.rst:224]:
```suggestion
The :guilabel:`RECYCLE RULES` sidebar lists each of the active recycle record rules, and displays the total
```
Upodating to match UI and Image

File: content/applications/productivity/data_cleaning.rst [hunk 3]
```diff
@@ -211,9 +211,117 @@ Click on the :icon:`fa-clone` :guilabel:`Duplicates` smart button to manage thes
...
+number of records detected beside each rule.
+
+By default, the :guilabel:`All` option is selected. Records are displayed in the list view, with the
+following columns:
+
+- :guilabel:`Record ID`: the ID of the original record.
+- :guilabel:`Record Name`: the name or title of the original record.
+
+Select a specific rule in the :guilabel:`RULE` sidebar to filter the duplicate records.
```
Comment (jero-odoo) [inline content/applications/productivity/data_cleaning.rst:233]:
```suggestion
Select a specific rule in the :guilabel:`RECYCLE RULES` sidebar to filter the duplicate records.
```

File: content/applications/productivity/data_cleaning.rst [hunk 4]
```diff
@@ -211,9 +211,117 @@ Click on the :icon:`fa-clone` :guilabel:`Duplicates` smart button to manage thes
...
+
+The :guilabel:`Recycle Records Rules` page (:menuselection:`Data Cleaning app --> Configuration -->
+Rules: Recycle Records`) is where the conditions for records to be recycled can be set.
+
+These rules can be configured for each model in the database, and with varying levels of
+specificity.
+
+.. tip::
+   The recycle rules run once every day, by default, as part of a scheduled action chron (*Data
+   Recycle: Clean Records*). However, each rule can be :ref:`ran manually
```
Comment (jero-odoo) [inline content/applications/productivity/data_cleaning.rst:261]:
```suggestion
   Recycle: Clean Records*). However, each rule can be :ref:`run manually
```

File: content/applications/productivity/data_cleaning.rst [hunk 5]
```diff
@@ -211,9 +211,117 @@ Click on the :icon:`fa-clone` :guilabel:`Duplicates` smart button to manage thes
...
+On the recycle record rule form, first choose a :guilabel:`Model` for this rule to target. Selecting
+a model updates the rule title to the chosen model.
+
+Optionally, configure a :guilabel:`Filter` to specify the records eligible for this rule. The number
+of eligible records is shown in the :icon:`oi-arrow-right` :guilabel:`# record(s)` link.
+
+Next, configure the field and time range for how the rule detects the records to recycle:
+
+- :guilabel:`Time Field`: select a field from the model to base the time (:dfn:`Delta`).
+- :guilabel:`Delta`: type the length of time, must be a whole number (e.g. `7`).
```
Comment (jero-odoo) [inline content/applications/productivity/data_cleaning.rst:275]:
```suggestion
- :guilabel:`Delta`: type the length of time, which must be a whole number (e.g., `7`).
```

**Review by jero-odoo — APPROVED** [review]
Hey @samueljlieber great work on this, just a few notes, otherwise this can move on to the next round :+1:

**Comment by samueljlieber** [comment]
Hi @ksc-odoo this PR is ready for a final review when you get a chance, thanks!

File: content/applications/productivity/data_cleaning.rst [hunk 6]
```diff
@@ -211,16 +211,123 @@ Click on the :icon:`fa-clone` :guilabel:`Duplicates` smart button to manage thes
...
+
+.. _data-cleaning/run-recycle-rule:
+
+Manually run a recycle rule
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To manually run a specific recycle rule at any time, navigate to :menuselection:`Data Cleaning app
+--> Configuration --> Rules: Recycle Records`, and select the rule to run.
+
+Then, on the rule form, select the :guilabel:`Run Now` button on the top-left. Upon doing so, the
```
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:320]:
```suggestion
Then, on the rule form, click the :guilabel:`Run Now` button on the top-left. Upon doing so, the
```
OPTIONAL: same reasoning as above re: "click" & "button" bias 😆 --- not required; just a suggestion.

File: content/applications/productivity/data_cleaning.rst [hunk 7]
```diff
@@ -211,16 +211,123 @@ Click on the :icon:`fa-clone` :guilabel:`Duplicates` smart button to manage thes
...
+
+Click the :icon:`fa-bars` :guilabel:`Records` smart button to manage these records.
 
 .. _data_cleaning/field-cleaning:
 
 Field cleaning
 ==============
 
-On the :guilabel:`Field Cleaning Records* dashboard (:menuselection:`Data Cleaning app --> Field
+On the :guilabel:`Field Cleaning Records dashboard` (:menuselection:`Data Cleaning app --> Field
```
Comment (ksc-odoo) [inline content/applications/productivity/data_cleaning.rst:330]:
```suggestion
On the :guilabel:`Field Cleaning Records` dashboard (:menuselection:`Data Cleaning app --> Field
```

**Review by ksc-odoo — APPROVED** [review]
Hey @samueljlieber -- just finished my Final Review of this PR. Fantastic job! Super well done. The majority of my comments are purely optional, so feel free to use whichever suggestions you agree with. Great work! Once you've implemented the changes, feel free to move this to the next stage. 👍

**Comment by samueljlieber** [comment]
Thank you for your very helpful review @ksc-odoo! Implemented your changes in https://github.com/odoo/documentation/commit/a7cb15fea52092b44060c675c5cae3638b206b7e .

**Review by StraubCreative — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**